ENGAGEHF/NavigationManager/NavigationManager.swift

```diff
@@ -17,23 +17,76 @@ import SwiftUI
 /// Navigation Manager
 ///
 /// Wraps an environment accessible and observable stack for use in navigating between views
+@MainActor
 @Observable
 class NavigationManager: Module, EnvironmentAccessible {
     @ObservationIgnored @Dependency private var configureFirebaseApp: ConfigureFirebaseApp
+    @ObservationIgnored @Dependency private var videoManager: VideoManager
     
     private let logger = Logger(subsystem: "ENGAGEHF", category: "NavigationManager")
     private var authStateDidChangeListenerHandle: AuthStateDidChangeListenerHandle?
     
-    var path = NavigationPath()
+    var educationPath = NavigationPath()
+    var medicationsPath = NavigationPath()
+    var heartHealthPath = NavigationPath()
+    var homePath = NavigationPath()
+    
+    var selectedTab: HomeView.Tabs = .home
     
     
     // On sign in, reinitialize to an empty navigation path
     func configure() {
         authStateDidChangeListenerHandle = Auth.auth().addStateDidChangeListener { [weak self] _, user in
             if user != nil {
                 self?.logger.debug("Reinitializing navigation path.")
-                self?.path = NavigationPath()
+                
+                self?.educationPath = NavigationPath()
+                self?.medicationsPath = NavigationPath()
+                self?.heartHealthPath = NavigationPath()
+                self?.homePath = NavigationPath()
+                
+                self?.selectedTab = .home
+            }
+        }
+    }
+    
+    
+    func execute(_ messageAction: MessageAction) async -> Bool {
+        self.logger.debug("Executing message action: \(messageAction.encodingString ?? "unknown")")
+        
+        switch messageAction {
+        case let .playVideo(sectionId, videoId):
+            let matchingVideo = videoManager.videoCollections
+                .filter { $0.id == sectionId }
+                .flatMap { $0.videos.filter { $0.id == videoId } }
+                .first
+            
+            guard let matchingVideo else {
+                self.logger.debug("No matching video found. sectionId: \(sectionId), videoId: \(videoId)")
+                return false
             }
+            
+            self.switchHomeTab(to: .education)
+            
+            try? await Task.sleep(for: .seconds(0.1))
+            
+            self.push(matchingVideo, onto: \.educationPath)
+        case .showMedications:
+            self.switchHomeTab(to: .medications)
+        case .showHeartHealth:
+            self.switchHomeTab(to: .heart)
+        default:
+            return false
         }
+        
+        return true
+    }
+    
+    func push<V: Hashable>(_ view: V, onto path: ReferenceWritableKeyPath<NavigationManager, NavigationPath>) {
+        self[keyPath: path].append(view)
+    }
```

Comment:
Wondering if we should rather make these dedicated function that directly take in specific types like a `Video` and then append it to the right path (we know at compile time what path it might be) to avoid appending an unsupported type to a navigation path.

Reply:
I think that makes a lot of sense @PSchmiedmayer! Just fyi, at the moment the only navigation stack we have an active path that actually has items appended to it is the one for the Education view. I will add in the rest of the push functions as the need arises and we determine what data types can be pushed onto that stack.

Reply:
Sounds great! 👍